web_crawler.py

```diff
@@ -0,0 +1,106 @@
+"""search for urls and save their html content to files asynchronously"""
+import asyncio
+import aiohttp
+import constants
+from bs4 import BeautifulSoup
+
+
+async def fetch(session, url):
+    """
+    fetch html format of a website asynchronously
+    :param session: the session to use for getting html
+    :param url: the url of the site
+    :return: html format of the website
+    """
+    async with session.get(url) as response:
+        html = await response.text()
+        return html
+
+
+async def fetch_all(urls):
+    """
+    fetch html formats of websites asynchronously
+    :param urls: urls of websites
+    :return: list of html formats of the websites
+    """
+    async with aiohttp.ClientSession() as session:
+        tasks = []
+        for url in urls:
+            tasks.append(fetch(session, url))
+        responses = await asyncio.gather(*tasks, return_exceptions=True)
+        return responses
+
+
+def main(url: list) -> None:
+    """
+    start a web crawler with a url given
+    :param url: url of the website
+    """
+    WebCrawler(url)
+
+
+def WebCrawler(urls: list) -> None:
+    """
+    asynchronously and recursively save html content of websites to files and find other urls
+    :param urls: list of the urls of website to crawl in
+    """
+    constants.urls_to_check = []
+    pages = asyncio.run(fetch_all(urls))
+    for i in range(len(pages)):
+        page = pages[i]
```

Comment:
You can do `for page in pages` and that would make `page` a different page everytime. You can also do `enumerate` and that would return the index + the page.